File: src/cosmic_pi_network/cvlib/cpp/ImageProcessing.cpp
#include "ImageProcessing.h"

cv::Mat ImageProcessing::apply_threshold(const cv::Mat& image) {
    cv::Mat thresh;
    cv::threshold(image, thresh, 0, 255, cv::THRESH_BINARY_INV + cv::THRESH_OTSU);
    return thresh;
}

cv::Mat ImageProcessing::apply_canny_edge_detection(const cv::Mat& image) {
    return cv::Canny(image, 50, 150);
}

File: src/cosmic_pi_network/cvlib/image_processing.py
import cv2
import numpy as np

class ImageProcessing:
    def __init__(self):
        pass

    def apply_threshold(self, image: np.ndarray) -> np.ndarray:
        _, thresh = cv2.threshold(image, 0, 255, cv2.THRESH_BINARY_INV + cv2.THRESH_OTSU)
        return thresh

    def apply_canny_edge_detection(self, image: np.ndarray) -> np.ndarray:
        return cv2.Canny(image, 50, 150)
